src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
```

Comment:
```suggestion
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```

Use `keyof` to typeguard function argument.